internal/pkg/core/plugin/process/status.go

```diff
@@ -101,9 +101,7 @@ func (a *Application) restart(proc *process.Info) {
 	a.stopWatcher(proc)
 
 	// kill the process
-	if proc != nil && proc.Process != nil {
-		_ = proc.Process.Kill()
-	}
+	_ = proc.Process.Kill()
```

Comment:
so, there is no way that the proc or the proc.Process can be nil here? 
seeing that the nil check was here before and now is removed.

Reply:
That seems indeed odd to me, I would keep that and separate investigation. See https://github.com/elastic/beats/pull/26126